chef_master/source/knife_setup.rst

```diff
@@ -56,7 +56,103 @@ Example config.rb:
+
+File paths, such as ``client_key`` or ``validator_key``, will be relative to ``~/.chef`` unless absolute paths are given. Clients can be identified with either ``node_name`` or ``client_name``, with ``client_name`` being preferred.
+
+Credentials for use with Target Mode (e.g. ``chef-client --target switch.example.org``) can also be stored as a separate profile in the credentials file. The name of the profile should match the DNS name of the target, and must be surrounded by single quotes when the name contains a period. For example: ``['switch.example.org']``. Keys that are valid configuration options will be passed to train, such as ``port``.
+
+.. code-block:: none
+
+   # Example .chef/credentials file
+   [default]
+   node_name = "barney"
+   client_key = "barney_rubble.pem"
+   chef_server_url = "https://api.chef.io/organisations/bedrock"
+
+   # a 'config context' such as knife can be is configured as a separate table
+   [default.knife]
+   ssh_user = 'ubuntu' # this would have been knife[:ssh_user] in your config.rb
+   aws_profile = 'engineering'
+   use_sudo = true
+
+   # a client_key may also be specified inline as in this example
+   [dev]
+   client_name = "admin"
+   client_key = """
+   -----BEGIN RSA PRIVATE KEY-----
+   MIICXAIBAAKBgQCqGKukO1De7zhZj6+H0qtjTkVxwTCpvKe4eCZ0FPqri0cb2JZfXJ/DgYSF6vUp
+   wmJG8wVQZKjeGcjDOL5UlsuusFncCzWBQ7RKNUSesmQRMSGkVb1/3j+skZ6UtW+5u09lHNsj6tQ5
+   1s1SPrCBkedbNf0Tp0GbMJDyR4e9T04ZZwIDAQABAoGAFijko56+qGyN8M0RVyaRAXz++xTqHBLh
+   3tx4VgMtrQ+WEgCjhoTwo23KMBAuJGSYnRmoBZM3lMfTKevIkAidPExvYCdm5dYq3XToLkkLv5L2
+   pIIVOFMDG+KESnAFV7l2c+cnzRMW0+b6f8mR1CJzZuxVxx6xx2fvLi55/mbSYxECQQDeAw6fiIQX
+   GukBI4eMZZt4nscy2o12KyYner3VpoeE+Np2q+Z3pvAMd/aNzQ/W9WaI+NRfcxUJrmfPwIGm63il
+   AkEAxCL5HQb2bQr4ByorcMWm/hEP2MZzROV73yF41hPsRC9m66KrheO9HPTJuo3/9s5p+sqGxOlF
+   L0NDt4SkosjgGwJAFklyR1uZ/wPJjj611cdBcztlPdqoxssQGnh85BzCj/u3WqBpE2vjvyyvyI5k
+   X6zk7S0ljKtt2jny2+00VsBerQJBAJGC1Mg5Oydo5NwD6BiROrPxGo2bpTbu/fhrT8ebHkTz2epl
+   U9VQQSQzY1oZMVX8i1m5WUTLPz2yLJIBQVdXqhMCQBGoiuSoSjafUhV7i1cEGpb88h5NBYZzWXGZ
+   37sJ5QsW+sJyoNde3xH8vdXhzU7eT82D6X/scw9RZz+/6rCJ4p0=
+   -----END RSA PRIVATE KEY-----
+   """   
+   validator_key = "test-validator.pem"
+   chef_server_url = "https://api.chef-server.dev/organizations/test"
+
+   ['web.preprod']
+   node_name = "brubble"
+   client_key = "preprod-brubble.pem"
+   chef_server_url = "https://preprod.chef-server.dev/organizations/preprod"
+
+   ['switch.example.org']
+   user = "cisco"
+   password = "cisco"
+   enable_password = "cisco"
+
+There are four ways to select which profile to use and are listed in priority order:
+
+    1. Pass the ``--profile`` option to knife, e.g. ``knife node list --profile dev``.
+    2. Set the profile name in the ``CHEF_PROFILE`` environment variable.
+    3. Write the profile name to the ``~/.chef/context`` file.
+    4. Otherwise, knife will use the 'default' profile.
+
+Knife Config
```

Comment:
@btm Would it be safe to say that this Knife Config deserves its own header/section?

Reply:
@mjingle yeah, it's not complete dependent on profiles, although most of it's functionality is profile related.